### PR TITLE
Don't smart indent code inside of Razor block constructs.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Prior to this change we weren't strict enough with where our smart indenter ran. We made the assumption that every code block could be smart indented and then Roslyn would "do the right thing". However, in nested code block scenarios we found that Roslyn and us would both indent resulting in extra newlines. These changes make the criteria for applying smart indentation a little stricter.
- Updated directive code block parsing to add a C# marker symbol in cases of an empty code block directive.
- Added unit tests to verify new smart indenter behavior.
- Updated existing tests to expect new syntax tree marker symbol for empty directive bits.
- Regenerated baselines.

image

@rynowak / @ajaybhargavb if you could review asap por favor 😄, time sensitive

#2410

### CODE REVIEW

File: test/Microsoft.VisualStudio.Editor.Razor.Test/BraceSmartIndenterTest.cs [hunk 1]
```diff
@@ -5,69 +5,267 @@
...
-            var result = BraceSmartIndenter.AtValidContentKind(changePosition, syntaxTree);
+            var result = BraceSmartIndenter.AtApplicableRazorBlock(changePosition, syntaxTree);
 
             // Assert
             Assert.False(result);
         }
 
         [Fact]
-        public void AtValidContentKind_ReturnsTrueAtCode()
+        public void AtApplicableRazorBlock_ReturnsTrueAtEplixitCodeBlocksCode()
```
Comment (ajaybhargavb) [inline test/Microsoft.VisualStudio.Editor.Razor.Test/BraceSmartIndenterTest.cs]: nit: Explicit

File: src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs [hunk 2]
```diff
@@ -242,35 +242,73 @@ internal static void RestoreCaretTo(int caretPosition, ITextView textView)
...
+            }
+
+            // Indentable content inside of a code block.
+            return true;
+        }
+
+        // Internal for testing
+        internal static bool ContainsInvalidContent(Span owner)
+        {
+            if (owner.Symbols.Count == 0)
```
Comment (ajaybhargavb) [inline src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs]: Why was this necessary? Why not indent empty code blocks as well? Please add a comment.

**Comment by ajaybhargavb** [comment]
Does this by any chance fix https://github.com/aspnet/Razor/issues/1076?

**Comment by NTaylorMullen** [comment]
> Does this by any chance fix #1076?

Sadly no. When the Razor smart indenter misbehaves it manifests as extra newlines.

File: src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs (hunk 2 above)
Comment (NTaylorMullen) [reply src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs]: Totally forget, gonna have to run through the scenarios again to re-verify.

Comment (NTaylorMullen) [reply src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs]: Has 0 value. Good catch 👍 👍 👍
